File: fix-imports.js
import fs from 'fs';
import path from 'path';
import { execSync } from 'child_process';

// Find all page.tsx files
function findPageFiles(dir) {
  const files = [];
  const items = fs.readdirSync(dir);
  
  for (const item of items) {
    const fullPath = path.join(dir, item);
    const stat = fs.statSync(fullPath);
    
    if (stat.isDirectory()) {
      files.push(...findPageFiles(fullPath));
    } else if (item === 'page.tsx') {
      files.push(fullPath);
    }
  }
  
  return files;
}

// Fix imports and component references in a file
function fixImports(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    const originalContent = content;
    
    // Fix ErrorBoundary import path
    content = content.replace(/import ErrorBoundary from '@\/components\/ErrorBoundary';/g, "import ErrorBoundary from '../components/ErrorBoundary';");
    content = content.replace(/import ErrorBoundary from '@\/components\/ErrorBoundary'/g, "import ErrorBoundary from '../components/ErrorBoundary'");
    
    // Fix PageComponent references
    content = content.replace(/<PageComponent \{\.\.\.props\} \/>/g, '<PageComponent {...props} />');
    
    // Fix function name if it's just 'Page'
    content = content.replace(/function Page\(\)/g, 'function PageComponent()');
    
    // Fix unused variable names
    content = content.replace(/const pageComponent =/g, 'const PageComponent =');
    content = content.replace(/pageComponent/g, 'PageComponent');
    
    // If we made changes, write the file back
    if (content !== originalContent) {
      fs.writeFileSync(filePath, content, 'utf8');
      console.log(`Fixed: ${filePath}`);
      return true;
    }
    
    return false;
  } catch (error) {
    console.error(`Error processing ${filePath}:`, error.message);
    return false;
  }
}

// Main function
function main() {
  console.log('Starting to fix imports...');
  
  const appDir = path.join(process.cwd(), 'app');
  const pageFiles = findPageFiles(appDir);
  
  console.log(`Found ${pageFiles.length} page files`);
  
  let fixedCount = 0;
  
  for (const filePath of pageFiles) {
    if (fixImports(filePath)) {
      fixedCount++;
    }
  }
  
  console.log(`Fixed ${fixedCount} files`);
  
  // Run type check to verify
  try {
    console.log('Running type check...');
    execSync('npm run type-check', { stdio: 'inherit' });
    console.log('Type check passed!');
  } catch (error) {
    console.error('Type check failed:', error.message);
  }
}

main();

File: fix_imports.py
#!/usr/bin/env python3
import os
import glob

def fix_import_paths(file_path):
    """Fix import paths for Navigation and Footer components"""
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # Check if file has incorrect import paths
        if '../components/Navigation' in content or '../components/Footer' in content:
            # Count directory levels to determine correct import path
            path_parts = file_path.split('/')
            app_index = path_parts.index('app')
            depth = len(path_parts) - app_index - 2  # -2 for 'app' and 'page.tsx'
            
            # Create correct import path
            import_path = '../' * depth + 'components/'
            
            # Replace import paths
            content = content.replace('../components/Navigation', f'{import_path}Navigation')
            content = content.replace('../components/Footer', f'{import_path}Footer')
            
            with open(file_path, 'w', encoding='utf-8') as f:
                f.write(content)
            
            return True
    except Exception as e:
        print(f"Error fixing {file_path}: {e}")
        return False

def main():
    # Find all page.tsx files in nested directories
    page_files = glob.glob('app/**/page.tsx', recursive=True)
    
    fixed_count = 0
    for file_path in page_files:
        # Only fix files in nested directories (not direct app/*/page.tsx)
        if file_path.count('/') > 2:
            if fix_import_paths(file_path):
                fixed_count += 1
    
    print(f"Fixed import paths in {fixed_count} page files")

if __name__ == "__main__":
    main()